Flakey test: JakartaWebSocketFrameHandler.OnMessageBinaryStreamTest

12.0.x

Getting failure on the `JakartaWebSocketFrameHandler.OnMessageBinaryStreamTest`:

> 
> java.lang.RuntimeException: Unexpected error
> 	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.0.26-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.java:78)
> 	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.0.26-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.performOnMessageInvocation(JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.java:44)
> 	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.0.26-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.testInvokeMessageStream(JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.java:70)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
> 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
> 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
> 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
> 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
> Caused by: java.util.concurrent.RejectedExecutionException: org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink$$Lambda$885/0x00000007c0281ba0@59c8fd8c
> 	at org.eclipse.jetty.util@12.0.26-SNAPSHOT/org.eclipse.jetty.util.thread.QueuedThreadPool.execute(QueuedThreadPool.java:782)
> 	at org.eclipse.jetty.websocket.core.common@12.0.26-SNAPSHOT/org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink.accept(DispatchedMessageSink.java:73)
> 	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.0.26-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.messages.AbstractDecodedMessageSink.accept(AbstractDecodedMessageSink.java:80)
> 	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.0.26-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandler.acceptMessage(JakartaWebSocketFrameHandler.java:589)
> 	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.0.26-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandler.onBinary(JakartaWebSocketFrameHandler.java:647)
> 	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.0.26-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandler.onFrame(JakartaWebSocketFrameHandler.java:243)
> 	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.0.26-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.java:74)
>

--- a/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -20,30 +20,36 @@ import jakarta.websocket.ClientEndpointConfig;
 import jakarta.websocket.EndpointConfig;
 import org.eclipse.jetty.ee10.websocket.jakarta.common.decoders.AvailableDecoders;
 import org.eclipse.jetty.ee10.websocket.jakarta.common.encoders.AvailableEncoders;
+import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.websocket.core.CoreSession;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
 public abstract class AbstractJakartaWebSocketFrameHandlerTest
 {
-    protected static DummyContainer container;
-    private static WebSocketComponents components;
+    protected DummyContainer container;
+    private WebSocketComponents components;
 
-    @BeforeAll
-    public static void initContainer() throws Exception
+    @BeforeEach
+    public void startContainer() throws Exception
     {
         container = new DummyContainer();
         container.start();
         components = new WebSocketComponents();
         components.start();
+
+        endpointConfig = ClientEndpointConfig.Builder.create().build();
+        encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
+        decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());
+        uriParams = new HashMap<>();
     }
 
-    @AfterAll
-    public static void stopContainer() throws Exception
+    @AfterEach
+    public void stopContainer()
     {
-        components.stop();
-        container.stop();
+        LifeCycle.stop(components);
+        LifeCycle.stop(container);
     }
 
     protected AvailableEncoders encoders;
@@ -52,21 +58,12 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
     protected EndpointConfig endpointConfig;
     protected CoreSession coreSession = new CoreSession.Empty()
     {
-
         @Override
         public WebSocketComponents getWebSocketComponents()
         {
             return components;
         }
     };
-
-    public AbstractJakartaWebSocketFrameHandlerTest()
-    {
-        endpointConfig = ClientEndpointConfig.Builder.create().build();
-        encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
-        decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());
-        uriParams = new HashMap<>();
-    }
 
     protected JakartaWebSocketFrameHandler newJakartaFrameHandler(Object websocket)
     {

--- a/jetty-ee9/jetty-ee9-websocket/jetty-ee9-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee9/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee9/jetty-ee9-websocket/jetty-ee9-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee9/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -20,30 +20,36 @@ import jakarta.websocket.ClientEndpointConfig;
 import jakarta.websocket.EndpointConfig;
 import org.eclipse.jetty.ee9.websocket.jakarta.common.decoders.AvailableDecoders;
 import org.eclipse.jetty.ee9.websocket.jakarta.common.encoders.AvailableEncoders;
+import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.websocket.core.CoreSession;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
 public abstract class AbstractJakartaWebSocketFrameHandlerTest
 {
-    protected static DummyContainer container;
-    private static WebSocketComponents components;
+    protected DummyContainer container;
+    private WebSocketComponents components;
 
-    @BeforeAll
-    public static void initContainer() throws Exception
+    @BeforeEach
+    public void startContainer() throws Exception
     {
         container = new DummyContainer();
         container.start();
         components = new WebSocketComponents();
         components.start();
+
+        endpointConfig = ClientEndpointConfig.Builder.create().build();
+        encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
+        decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());
+        uriParams = new HashMap<>();
     }
 
-    @AfterAll
-    public static void stopContainer() throws Exception
+    @AfterEach
+    public void stopContainer()
     {
-        components.stop();
-        container.stop();
+        LifeCycle.stop(components);
+        LifeCycle.stop(container);
     }
 
     protected AvailableEncoders encoders;
@@ -52,21 +58,12 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
     protected EndpointConfig endpointConfig;
     protected CoreSession coreSession = new CoreSession.Empty()
     {
-
         @Override
         public WebSocketComponents getWebSocketComponents()
         {
             return components;
         }
     };
-
-    public AbstractJakartaWebSocketFrameHandlerTest()
-    {
-        endpointConfig = ClientEndpointConfig.Builder.create().build();
-        encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
-        decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());
-        uriParams = new HashMap<>();
-    }
 
     protected JakartaWebSocketFrameHandler newJakartaFrameHandler(Object websocket)
     {

